marklogic-data-hub/src/test/java/com/marklogic/hub/flow/impl/FlowRunnerTest.java

```diff
@@ -81,6 +84,89 @@ public void testRunFlow() {
         Assertions.assertTrue(file.contains("ingest.csv"));
     }
 
+    /*
+     https://bugtrack.marklogic.com/54003
+     When stemmed search is set to "advanced", fetching step definition fails and hence running flow would fail in all server versions.
+     After that if db is re-indexed, fetching flow succeeds in versions < 10.0-4 and fails in version >=10.0-5 and nightly server.
+     With this server fix, fetching step definition, legacy mappings (and other artifacts ?) would fail and the json property value
+     query requires 'lang=zxx' to be included in order to ensure artifacts can be fetched if stemming (and/or reindexing is done) is turned on.
+     */
+    @Test
+    void runFlowWithStemmingSearch() throws InterruptedException {
+        Versions.MarkLogicVersion mlVersion = versions.getMLVersion();
+        Assumptions.assumeTrue(mlVersion.getMajor() >=10);
+        HubConfig hubConfig = runAsDataHubDeveloper();
+        try{
+            updateStemmingOnStagingDb(hubConfig,true);
+            RunFlowResponse resp = runFlow();
+            Assertions.assertTrue(JobStatus.STOP_ON_ERROR.toString().equalsIgnoreCase(resp.getJobStatus()));
+            hubConfig = runAsDataHubDeveloper();
+            DatabaseManager dbManager = new DatabaseManager(hubConfig.getManageClient());
+            dbManager.reindexDatabase(hubConfig.getDbName(DatabaseKind.STAGING));
+            String query = "(\n" +
+                "  for $forest-id in xdmp:database-forests(xdmp:database('data-hub-STAGING'))\n" +
+                "  return xdmp:forest-status($forest-id)//*:reindexing\n" +
+                ") = fn:true()";
+            StringHandle stringHandle;
+            String status;
+            int attempts = 5;
+            do{
+                Thread.sleep(5000L);
```

Comment:
You can use "sleep" in the parent class so you don't have to deal with InterruptedException.

Reply:
Also, let's get this all into a separate method called "waitForReindexingToFinish". And limit the sleep to maybe 500ms or so - the call to check on it is extremely fast, so we want this loop to finish as soon as the reindexer is done.